### PR TITLE
feat: add hashing api to wallet secret keys

### PR DESCRIPTION
**Pull request by hansieodendaal**
Description
---
Partial fulfillment for #4395:
- base_layer/wallet/src/output_manager_service/service.rs, fn hash_secret_key(...)
- base_layer/wallet/src/transaction_service/service.rs, fn hash_secret_key(...)

Motivation and Context
---
Domain separated hashing as per the `tari_crypto` hashing API is needed for secret keys 

How Has This Been Tested?
---
Unit tests pass
Cucumber tests pass

### CODE REVIEW

File: applications/tari_console_wallet/src/main.rs [hunk 1]
```diff
@@ -100,7 +100,7 @@ fn main_inner() -> Result<(), ExitError> {
     )?;
 
     #[cfg_attr(feature = "libtor", allow(unused_mut))]
-    let mut config = ApplicationConfig::load_from(&cfg)?;
+    let config = ApplicationConfig::load_from(&cfg)?;
```
Comment (sdbondi) [inline applications/tari_console_wallet/src/main.rs]: This mut is used by libtor
Clippy fix here: https://github.com/tari-project/tari/pull/4425

**Review by sdbondi — DISMISSED** [review]

File: base_layer/wallet/src/lib.rs [hunk 2]
```diff
@@ -54,3 +55,10 @@ pub type WalletSqlite = Wallet<
     ContactsServiceSqliteDatabase,
     KeyManagerSqliteDatabase,
 >;
+
+hash_domain!(
+    WalletSecretKeysDomain,
+    "com.tari.tari_project.base_layer.wallet.secret_keys",
```
Comment (sdbondi) [inline base_layer/wallet/src/lib.rs:61]: Why add `tari_project` to the hash domain?

File: applications/tari_console_wallet/src/main.rs (hunk 1 above)
Comment (hansieodendaal) [reply applications/tari_console_wallet/src/main.rs]: Thanks, implemented

File: base_layer/wallet/src/lib.rs (hunk 2 above)
Comment (hansieodendaal) [reply base_layer/wallet/src/lib.rs:61]: This is to differentiate between different Tari GitHub projects.

**Review by stringhandler — APPROVED** [review]

**Pull request merged**